src/Storages/MergeTree/KeyCondition.cpp

```diff
@@ -1036,6 +1039,24 @@ bool applyFunctionChainToColumn(
     return true;
 }
 
+bool KeyCondition::isFunctionReallyMonotonic(const IFunctionBase & func, const IDataType & arg_type) const
```

Comment:
This looks like it should be part of `FunctionConvert::hasInformationAboutMonotonicity()`  or even somehow inside `ToDateTimeMonotonicity`, but not not a custom function in KeyCondition